jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/CloseStatus.java

```diff
@@ -324,4 +334,27 @@ public String toString()
         return String.format("{%04d=%s,%s}", code, codeString(code), reason);
     }
 
+    public interface Supplier
```

Comment:
Ah that was the intention.... only have refactored.

Reply:
That was the intention... but because we have to cast to Supplier, the generic doesn't work well because you can't cast to a typed generic.  So this is a Supplier by nature, but a well typed one - so I think it works OK.   But then the cast is only in a single utility method....